cmd/uql/execute.go

```diff
@@ -96,7 +96,7 @@ type position struct {
 
 type uqlService interface {
 	Execute(query *Query, apiVersion ApiVersion) (parsedResponse, error)
-	Continue(link *Link) (parsedResponse, error)
+	Continue(link *Link, options *api.Options) (parsedResponse, error)
```

Comment:
**uqlService** does not really care about **options** value. It's just passed down to **api.JSONPost** method. Because the **api** namespace method is static, it cannot be configured prior. The same issue is with **uql.ExecuteQuery** and **uql.ContinueQuery** methods. I suggest moving these two methods to a client interface. The **uql** namespace could have statical instance. Add **api.Options** to the existing **type defaultBackend struct** and initialize your own new uql client with modified **defaultBackend** with our desired options

Reply:
@ace-timr hopefully this is what you had in mind with this suggestion but do let me know if further alterations are required

https://github.com/cisco-open/fsoc/blob/e3338c589daffddf0f80facb966729c53d87cad0/cmd/uql/execute.go#L147-L175